httpclient/httpclient.go

```diff
@@ -740,3 +753,29 @@ func doneRetrying(err error) error {
 	}
 	return err
 }
+
+func (r Request) validate() (err error) {
+	// We do not allow GET requests with Body as they are not supported by CloudFront WAF, which requests are routed
+	// through. - https://circleci.slack.com/archives/C03M4P0Q4GH/p1659566842825159
+	if r.validateGetWithBody() {
+		return errors.New("cannot have GET request with body or raw body")
```

Comment:
Should we make mention of the allowGETWithBody flag not being set here as well? If not in the error (impl detail) in the comment above.

Reply:
I think that makes sense.